scanner/redhatbase.go

```diff
@@ -699,20 +711,27 @@ func splitFileName(filename string) (name, ver, rel, epoch, arch string, err err
 	basename := strings.TrimSuffix(filename, ".rpm")
 
 	archIndex := strings.LastIndex(basename, ".")
+	// support not standard style rpm fullname
+	// e.g.
+	//   baz-0-1-i386 => i386
+	//   qux-0--i386 => i386
+	if i := strings.LastIndex(basename[archIndex+1:], "-"); i > -1 {
+		archIndex = archIndex + (i + 1)
+	}
 	if archIndex == -1 {
-		return "", "", "", "", "", xerrors.Errorf("unexpected file name. expected: %q, actual: %q", "<name>-<version>-<release>.<arch>.rpm", fmt.Sprintf("%s.rpm", filename))
+		return "", "", "", "", "", xerrors.Errorf("unexpected file name. expected: %q, actual: %q", "(<epoch>:)<name>-<version>-(<release>)(.|-)<arch>.rpm", fmt.Sprintf("%s.rpm", filename))
```

Comment:
filename includes ".rpm" suffix, so fmt.Sprintf() for concatenation seems to be unnecessary.
Sorry that this bug is introduced not by this PR, but my previous PR.

Reply:
fixed: https://github.com/future-architect/vuls/compare/8a0ddb0f8c2e72816772156338d1d454d44f68a9..720b91407824f641e60909e2a714a1f4cea19649

Reply:
Thanks a lot!